### PR TITLE
Pass updated replacementvariables when measuring progress

### PR DESCRIPTION
**Pull request by maartenleenders**
## Summary

This PR can be summarized in the following changelog entry:

* Changes in replacevariables now cause progress in SEO Title and description to be measured with the new replacevariable values.

## Relevant technical choices:

* I chose to optionally pass replacementVariables to `mapDataToMeasurements` (and eventually `processReplacementVariables`), so that when mapDataToMeasurements is accessed from the `componentWillReceiveProps` function, the "next" replacementVariables can be used.

## Test instructions

This PR can be tested by following these steps:

* Put `%%title%%` in the SEO Title input field.
* Change the title of the post. See that the progress changes accordingly.
* Do the same for the description (put `%%title%%` in it and change the title).
* It should work for any replacement variable that you change, and that is placed in the title or description field.

Fixes https://github.com/Yoast/wordpress-seo/issues/10047

### CODE REVIEW

File: composites/Plugin/SnippetEditor/components/SnippetEditor.js [hunk 1]
```diff
@@ -146,21 +146,27 @@ class SnippetEditor extends React.Component {
...
+	 * Returns whether the old and the new data or replacement variables are different.
 	 *
-	 * @param {Object} oldData The old data.
-	 * @param {Object} newData The new data.
+	 * @param {Object} prev The old props.
+	 * @param {Object} next The new props.
 	 * @returns {boolean} True if any of the data points has changed.
 	 */
-	shallowCompareData( oldData, newData ) {
+	shallowCompareData( prev, next ) {
```
Comment (atimmer) [inline composites/Plugin/SnippetEditor/components/SnippetEditor.js]: I meant to call this `prevProps` and `nextProps`.

**Comment by boblinthorst** [comment]
CR: ok 👍

**Comment by Xyfi** [comment]
Acceptance :construction: 

If you paste a title it now updates properly, but when you remove the entire title when selecting it and deleting it, it does not. Also when holding backspace until the title field is empty the progress bar jumps from empty to filled when the last character is removed from the title field.

title-measurement-error

**Comment by danielFangstrom** [comment]
The issue is that when the value of the replacement variable is empty, it gets the `%%title%%` as a replacement. This takes the value from rawdata in the app, which is not updated by the same events. This is probably not the main issue though as the value is later correctly updated, but the progress bar is not updated. There probably needs to be an additional trigger for the progress update.

**Comment by boblinthorst** [comment]
Merging for testing purposes: seperate issue will be made for the updating issue still present.

**Pull request merged**